File: src/util/SqlServerUtil.scala
/*
package util

import java.sql.{DriverManager, ResultSet}

/**
  * sqlServer数据库的连接帮助类
  */
object SqlServerUtil {
  private final val SQL_SERVER_CONN_URL = PropertiesUtil.getValueFromConf("sqlserver.connect.url")

  def insert(sql: String, params: Array[String]): Int = {
    var res = 0
    Class.forName("com.microsoft.sqlserver.jdbc.SQLServerDriver").newInstance()
    val conn = DriverManager.getConnection(SQL_SERVER_CONN_URL)
    try {
      // Configure to be Read Only
      val statement = conn.prepareStatement(sql)
      for (i <- 0 to params.length - 1) {
        val j = i + 1
        statement.setString(j, params(i))
      }
      res = statement.executeUpdate()
    }
    catch {
      case e: Exception => {
        e.printStackTrace()
      }
    } finally {
      conn.close()
    }
    res
  }

  def query(sql: String, colName: String): List[String] = {
    //classOf[com.mysql.jdbc.Driver]
    Class.forName("com.microsoft.sqlserver.jdbc.SQLServerDriver").newInstance();
    val conn = DriverManager.getConnection(SQL_SERVER_CONN_URL)
    var resultList: List[String] = List()
    // println("hello")
    try {
      // Configure to be Read Only
      val statement = conn.createStatement(ResultSet.TYPE_FORWARD_ONLY, ResultSet.CONCUR_READ_ONLY)

      // Execute Query
      val rs: ResultSet = statement.executeQuery(sql)

      // Iterate Over ResultSet
      while (rs.next) {
        resultList = resultList ++ List(rs.getString(colName))
        println(rs.getString(colName))
      }

    }
    catch {
      case _: Exception => println("===>")
    }
    finally {
      conn.close
    }
    resultList
  }

  def getLatestSuccessDay(sql: String): String ={    //  banner=R10003 , data_day最近，而且 log_level全部非error。
    Class.forName("com.microsoft.sqlserver.jdbc.SQLServerDriver").newInstance()
    val conn = DriverManager.getConnection(SQL_SERVER_CONN_URL)
    var latestSuccessDay: String =""
    // println("hello")
    try {
      // Configure to be Read Only
      val statement = conn.createStatement(ResultSet.TYPE_FORWARD_ONLY, ResultSet.CONCUR_READ_ONLY)

      // Execute Query
      val rs: ResultSet = statement.executeQuery(sql)

      // Iterate Over ResultSet
      while (rs.next) {
        resultList = resultList ++ List(rs.getString(colName))
        println(rs.getString(colName))
      }

    }
    catch {
      case _: Exception => println("===>")
    }
    finally {
      conn.close
    }
    latestSuccessDay
  }

}
*/

File: src/util/SqlServerUtil.java
package util;

import java.sql.*;
import java.util.ArrayList;
import java.util.List;

/**
 * Created by SAM on 2018/5/10.
 */
public class SqlServerUtil {
    public static PropertiesUtil properties = new PropertiesUtil();
    private static final String SQL_SERVER_CONN_URL = properties.getProperty("sqlserver.connect.url");
    static Connection conn;
   public static int  insert(String sql, List<String> params) {   //插入语句。
        int res = 0;
        try {
            Class.forName("com.microsoft.sqlserver.jdbc.SQLServerDriver").newInstance();
            conn = DriverManager.getConnection(SQL_SERVER_CONN_URL);
            // Configure to be Read Only
            PreparedStatement statement = conn.prepareStatement(sql);
            //P6PreparedStatement p6stmt = new P6PreparedStatement(null, statement, null, sql);  //test 显示sql.
            for (int i= 0 ; i< params.size(); i++ ) {
                int j = i + 1;
                statement.setString(j, params.get(i));   //把字符串列表内的字符串全部导入sql内的参数。
            }
            res = statement.executeUpdate();
        } catch (Exception e) {
            e.printStackTrace();
        } finally {
            try{
                conn.close();
            }catch (Exception e) {
                e.printStackTrace();
            }
        }
        return  res;
    }

    public static List<String> getLatestDay(String sql, String colName) {   //JDBC查询数据库语句。 执行sql语句，获取ResultSet， resultList是ResultSet中某个列的值的list.
        //String LatestFailureDay = "";
        List<String> resultList = new ArrayList<String>();
        try {
            Class.forName("com.microsoft.sqlserver.jdbc.SQLServerDriver").newInstance();
            Connection conn = DriverManager.getConnection(SQL_SERVER_CONN_URL);
            // Configure to be Read Only
            Statement statement = conn.createStatement(ResultSet.TYPE_FORWARD_ONLY, ResultSet.CONCUR_READ_ONLY);
            // Execute Query
            ResultSet rs = statement.executeQuery(sql);
            // Iterate Over ResultSet
            while (rs.next()) {
                resultList.add(rs.getString(colName));
                //LatestFailureDay=rs.getString(colName);
                //System.out.println(rs.getString(colName));
            }
        } catch (Exception e) {
            e.printStackTrace();
        } finally {
            try{
                if(conn != null)
                conn.close();  //????
            }catch (Exception e) {
                e.printStackTrace();
            }
        }
        return  resultList;
    }
}
